background/lib/simple-hash_update.ts

```diff
@@ -0,0 +1,262 @@
+
+  const thumbnail =
+    previewURL ||
+    previews?.image_large_url ||
+    previews?.image_medium_url ||
+    previews?.image_small_url
+  const chainID = SIMPLE_HASH_CHAIN_TO_ID[chain]
+
+  const transferDate = owners.find(({ owner_address }) =>
+    sameEVMAddress(owner_address, owner)
+  )?.last_acquired_date
+
+  const attributes =
+    metadata?.attributes?.map(({ trait_type, value }) => ({
+      value,
+      trait: trait_type,
+    })) ?? []
+
+  return {
+    id: nftID,
+    name,
+    description,
+    thumbnail,
+    transferDate,
+    attributes,
+    collectionID,
+    contract: contractAddress,
+    owner,
+    network: NETWORK_BY_CHAIN_ID[chainID],
+    achievement: isAchievement ? { url: nftURL } : null,
+  }
+}
+
+/**
+ * Get NFT holdings of given address across collections and networks
+ * using the SimpleHash API.
+ *
+ * Learn more at https://simplehash.readme.io/reference/nfts-by-owners
+ *
+ * @param address address whose NFT holdings we want to query
+ * @param collectionID collections we are updating
+ * @param chainIDs the networks we're querying
+ */
+export async function getNFTs(
+  address: string,
+  collectionID: string,
+  chainIDs: string[]
+): Promise<NFTsWithPagesResponse> {
+  const requestURL = new URL("https://api.simplehash.com/api/v0/nfts/owners")
+  requestURL.searchParams.set("chains", getChainIDsNames(chainIDs))
+  requestURL.searchParams.set("wallet_addresses", address)
+  requestURL.searchParams.set("collection_id", collectionID)
+
+  const headers = new Headers()
+  headers.set("X-API-KEY", process.env.SIMPLE_HASH_API_KEY ?? "")
+
+  try {
+    const result = (await (
+      await fetchWithTimeout(requestURL.toString(), {
+        headers,
+      })
+    ).json()) as unknown as SimpleHashNFTsByWalletAPIResponse
+
+    return {
+      nfts:
+        result.nfts
+          .filter((nft) => !!nft.nft_id)
+          .map((nft) => simpleHashNFTModelToNFT(nft, address)) ?? [],
+      nextPageURL: result.next,
+    }
+  } catch (err) {
+    logger.error("Error retrieving NFTs ", err)
+  }
+
+  return { nfts: [], nextPageURL: null }
+}
+
+/**
+ * Get NFT Collections of given address and networks using the SimpleHash API.
+ * This will return an overview of collections that address holds.
+ *
+ * Learn more at https://simplehash.readme.io/reference/collections-by-wallets
+ *
+ * @param address address whose NFT Collections we want to query
+ * @param chainIDs the networks we're querying
+ */
+export async function getCollections(
+  address: string,
+  chainIDs: string[]
+): Promise<NFTCollection[]> {
+  const requestURL = new URL(
+    "https://api.simplehash.com/api/v0/nfts/collections_by_wallets"
+  )
+  requestURL.searchParams.set("chains", getChainIDsNames(chainIDs))
+  requestURL.searchParams.set("wallet_addresses", address)
+
+  const headers = new Headers()
+  headers.set("X-API-KEY", process.env.SIMPLE_HASH_API_KEY ?? "")
+
+  try {
+    const result = (await (
```

Comment:
We can also avoid type assertions here
```suggestion
    const result: SimpleHashCollectionsByWalletAPIResponse = (await (
```